From 2f08c29dcd22b576b6999b1aa0e790049bb9dc2b Mon Sep 17 00:00:00 2001
From: Jiangjin Wang <kaymw@aosc.io>
Date: Sun, 22 Oct 2023 22:13:17 -0700
Subject: [PATCH 12/18] BACKPORT: LOONGARCHLINUX: Add support for LoongArch64

Adapted from LoongArchLinux. Rebased to Firefox 118.0.2.

Co-Authored-By: loongson <loongson@loongson.cn>
Co-Authored-By: WANG Xuerui <xen0n@gentoo.org>

Link: https://github.com/loongarchlinux/extra/blob/2024.08/firefox/0001-Add-support-for-LoongArch64.patch
[Kexy: Resolved minor conflict in
 toolkit/components/telemetry/pingsender/pingsender.cpp]
Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 third_party/libwebrtc/build/build_config.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/third_party/libwebrtc/build/build_config.h b/third_party/libwebrtc/build/build_config.h
index c39ae9da50..28191de026 100644
--- a/third_party/libwebrtc/build/build_config.h
+++ b/third_party/libwebrtc/build/build_config.h
@@ -210,6 +210,10 @@
 #define ARCH_CPU_SPARC 1
 #define ARCH_CPU_32_BITS 1
 #define ARCH_CPU_BIG_ENDIAN 1
+#elif defined(__loongarch_lp64)
+#define ARCH_CPU_LOONGARCH64 1
+#define ARCH_CPU_64_BITS 1
+#define ARCH_CPU_LITTLE_ENDIAN 1
 #else
 #error Please add support for your architecture in build/build_config.h
 #endif
-- 
2.47.0

